content/chainguard/chainguard-images/getting-started/c/index.md

```diff
@@ -0,0 +1,333 @@
+
+## Example 3 --- Multi-Stage Build for C++ Applications
+
+So far, our demonstrations have featured a program coded in C. A similar image building process applies to binaries compiled for the C++ programming language. 
+
+### Step 1: Setting up a Demo Application
+
+In your terminal, create a new file called `hello.cpp` with the text editor of your choice.
+
+```sh
+nano hello.cpp
+```
+
+Copy the following C++ code to the file you just created. This code will display a simple greeting specifying that it was written in C++.
+
+```C++
+/* Chainguard Academy (edu.chainguard.dev)
+*  Getting Started with the C/C++ Chainguard Images
+*  Example 3 - C++
+*/
+
+#include <iostream>
+using namespace std;
+
+// Main Function
+int main(){
+    cout << "Hello, world!\n";
+    cout << "I am a demo from the Chainguard Academy.\n";
+    cout << "My code was written in C++.\n";
+
+    return 0;
+}
+```
+
+When you are done editing your file, save and close it.
+
+You can now compile your C++ program using `g++`. Execute the following command in your terminal to compile the program. The command will display any compiler warnings or errors and will name the resultant binary `hello`.
+
+```sh
+g++ -Wall -o hello hello.cpp
+```
+
+Now you can test your compiled binary.
+
+```sh
+./hello
+```
+
+You should see the following output in your terminal.
+
+```Output
+Hello, world!
+I am a demo from the Chainguard Academy.
+My code was written in C++.
+```
+
+### Step 2: Creating the Dockerfile
+
+Now, with a working C++ example, we can compile and run our program using a multi-stage build. With the text editor of your choice, create a new file named `Dockerfile3`.
+
+```sh
+nano Dockerfile3
+```
+
+The following Dockerfile will:
+1. Use the `gcc-glibc` Chainguard Image as the builder stage;
+2. Set the current working directory of the image to `/usr/bin`;
+3. Copy our `hello.cpp` program code to the current directory;
+4. Compile the program using `g++` and name it `hello`;
+5. Begin a new stage using the `glibc-dynamic` Chainguard Image;
+6. Set the current working directory of the new image to `/usr/bin`;
+7. Copy the compiled `hello` C++ binary from our builder stage;
+8. Execute our binary from the `glibc-dynamic` image when the container is started.
+
+```Dockerfile3
+# Example 3 - Multi-Stage Build for C++
+
+FROM cgr.dev/chainguard/gcc-glibc:latest AS builder
+
+WORKDIR /usr/bin
+
+COPY hello.cpp ./
+
+RUN ["g++", "-Wall", "-o", "hello", "hello.cpp"]
+
+FROM cgr.dev/chainguard/glibc-dynamic:latest
+
+WORKDIR /usr/bin
+
+COPY --from=builder /usr/bin/hello ./
+
+ENTRYPOINT ["./hello"]
+```
+
+When you are finished editing your Dockerfile, save and close it.
+
+With our new Dockerfile created, we can build the image. Execute the following command in your terminal to build your multi-stage C++ image.
+
+```sh
+docker build -f Dockerfile3 -t example3:latest .
+```
+
+With your image built, we can now run it with the following command.
```

Comment:
```suggestion
With your image built, you can now run it with the following command.
```

I feel like I might have missed some paragraphs that split between "you" and "we" - maybe do a scan for this?